### PR TITLE
[ssh] Bug Fix: Change how ssh vm reads the ssh client logs

### PR DESCRIPTION
**Pull request by vthiebaut10**
Fix: https://github.com/Azure/azure-cli-extensions/issues/4915#issuecomment-1243939019

- ssh vm reads the logs from ssh client and ssh proxy to delete sensitive information when it is no longer needed and print helpful error messages for known errors. Our approach to do that was causing ssh banners to be printed after the ssh process was terminated, and not before authentication. 
- Known issue: we are no longer able to read ssh proxy error channel when the user is running the command from a linux machine to connect to a local user on an arc machine. That is not a breaking change, but users in this particular situation will not get a helpful error message if there is an error with the proxy. 
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
ssh

File: src/ssh/HISTORY.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 Release History
 ===============
+1.1.3
+-----
+* Fix bug where extension prints ssh banners after the connection is closed.
```
Comment (zhoxing-ms) [inline src/ssh/HISTORY.md:5]: If you want to release a new extension version, please also upgrade the version defined in `setup.py`

Comment (vthiebaut10) [reply src/ssh/HISTORY.md:5]: @zhoxing-ms I was planning on upgrading the version only after all the open PRs for the next release are merged and I have the chance to run some final tests. For now I'll just remove the changes in the HISTORY.md file for this PR.

Comment (zhoxing-ms) [reply src/ssh/HISTORY.md:5]: @vthiebaut10 OK. Or I can merge this PR first, and you can add new content to the history notes of subsequent PRs.

Comment (zhoxing-ms) [reply src/ssh/HISTORY.md:5]: In the future, if you do not want to release a new version immediately, I suggest you add comments in PR to tell us

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**